app/controllers/projects_controller.rb

```diff
@@ -106,7 +115,22 @@ def get_current_stories
     @stories ||= []
   end
 
+  def populate_languages_dropdown
+    @projects_languages_array = Array.new
+    Language.all.each do |language|
+      @projects_languages_array << language.name
+    end
+  end
+
+  def filter_projects
+    @language = params[:project][:languages]
+    @filtered_projects = Project.search_by_language(@language, params[:page])
```

Comment:
I think the rule of thumb here is to avoid starting a new query with a `Project.` if we can chain on to the existing query and vary it according to circumstances that will generally be smoother - although now always :-)

Reply:
Are you suggesting that I change this @tansaku or should I go with the implementation you used :arrow_up: and make sure it works for me and passes all the tests?

Reply:
@mattwr18 - up to you - I think I fixed the issue with the pagination working across the search by language (although I might be fooling myself) - and you should totally feel free to use my code/fix.

but if you want to re-implement, or use part of it, or use it and re-factor that's all good too.  Depends on your energy level and what's best for your learning ...

I guess technically we should have a test that shows search by language working across the pagination - although I can see how that might seem redundant if we're going to replace with infinite scroll soon - although of course at that time the test should evolve into checking that form of pagination ...

Am I making any sense? :-) decision in your hands of course :-)

Reply:
Judging from your emoji response, and re-reading my comment, I think I didn't express what I was trying to ask. I was a little unclear how to proceed with getting this in, and briefly thought that maybe you would just commit your changes. 

I think it is better for me to do it though. There are still a couple things I don't like though. 
   ~~1. It seems like I needed to add a line `@project = Project.new` to the index page to get the 
       languages dropdown to work. It seems you didn't need to do this~~ Actually, just figures this 
       out.
       2. The other is that the index method is now 17 lines!! 4 lines are to set the initial @projects 
       which could be refactored into a private method in the ProjectsContoller? 4 lines are the 
       respond_to block that I'm not aware of a way to refactor.  Should the refactoring be a 
       separate ticket?

I have added test coverage for search by language working across the pagination.